src/Badges.sol

```diff
@@ -66,14 +72,25 @@ contract Badges is
       super.supportsInterface(_interfaceId);
   }
 
-  // permissionInt:
-  // 0: only raft owner
-  // 1: all badge holders associated with that RAFT
-  // 2: only specific badge holders
-  function setBadgeSpecCreationPermissions(uint256 raftTokenId, uint256 permissionInt) external {
-    // this function will only be called when a DAO is created
-    // would it be helpful in any way to emit an event here?
-    raftToBadgeSpecCreationPermissions[raftTokenId] = permissionInt;
+  // this is a org-level function that's called from the settings page on a DAO
+  function setSpecCreationPermissions(
+    uint256 _raftTokenId,
+    Permission permissionInt,
+    string[] memory listOfSpecs
+  ) external {
+    // require that the caller is the owner of the raftTokenId
+    require(specDataHolder.getRaftOwner(_raftTokenId) == msg.sender, "setSpecCreationPermissions: unauthorized");
+
+    // would it be helpful to emit an event here?
```

Comment:
I think all write actions in general should emit an event so we can index these.